eng/pipelines/templates/Tests_On_Linux.yml

```diff
@@ -0,0 +1,72 @@
+steps:
+- task: ShellScript@2
+  displayName: "Install .NET 5.0 for build"
+  continueOnError: "true"
+  inputs:
+    scriptPath: "scripts/utils/InstallCLIforBuild.sh"
+    disableAutoCwd: "true"
+    cwd: "$(Build.Repository.LocalPath)"
+    args: '$(SDKVersionForBuild)'
+
+- task: PowerShell@2
+  displayName: "Define variables"
+  inputs:
+    targetType: "inline"
+    script: |
+      Write-Host "##vso[task.setvariable variable=PATH]${env:AGENT_TEMPDIRECTORY}/dotnet/:${env:PATH}"
+
+- task: PowerShell@2
+  displayName: "Update Build Number"
+  inputs:
+    targetType: "inline"
+    script: |
+      Write-Host "##vso[build.updatebuildnumber]$env:FULLVSTSBUILDNUMBER"
+    failOnStderr: "true"
+  condition: "always()"
+
+- task: ShellScript@2
+  displayName: "Run Tests (continue on error)"
+  continueOnError: "true"
+  inputs:
+    scriptPath: "scripts/funcTests/runFuncTests.sh"
+    disableAutoCwd: "true"
+    cwd: "$(Build.Repository.LocalPath)"
+  condition: "and(succeeded(), eq(variables['IsOfficialBuild'], 'true'))"
+
+- task: ShellScript@2
+  displayName: "Run Tests (stop on error)"
+  continueOnError: "false"
+  inputs:
+    scriptPath: "scripts/funcTests/runFuncTests.sh"
+    disableAutoCwd: "true"
+    cwd: "$(Build.Repository.LocalPath)"
+  condition: "and(succeeded(), not(eq(variables['IsOfficialBuild'], 'true')))"
+
+- task: PublishTestResults@2
+  displayName: "Publish Test Results"
+  inputs:
+    testRunner: "XUnit"
+    testResultsFiles: "*.xml"
+    testRunTitle: "NuGet.Client Tests On Linux"
```

Comment:
Comparing `eng/pipelines/templates/Tests_On_Linux.yml` with `eng/pipelines/templates/Tests_On_Mac.yml` looks like `testRunTitle` will vary based on the OS.  Can we some how use same `.yml` for `Linux` and `Mac` Tests. The same comment applies to `Line no:70` in this `.yml` file.

Reply:
`MacTests.yml` file has additional `"Download NuGet.CommandLine.Test artifacts"` task compare to `LinuxTests.yml`

Reply:
yes we can, but it's outside of the scope of this PR, which is just to split them into more managable pieces. In fact, this split is what makes the comparison of the Mac/Linux jobs so easy. I'd like to get this split in ASAP, so we can start making other improvements with less risk of merge conflicts.